pkg/proxy/util/iptables/traffic.go

```diff
@@ -87,3 +87,62 @@ func (d *detectLocalByCIDR) IfLocal() []string {
 func (d *detectLocalByCIDR) IfNotLocal() []string {
 	return d.ifNotLocal
 }
+
+type detectLocalByBridgeInterfaceName struct {
+	ifLocal    []string
+	ifNotLocal []string
+}
+
+// NewDetectLocalByBridgeInterfaceName implements the LocalTrafficDetector interface using a bridge interface name.
+// This can be used when a bridge can be used to capture the notion of local traffic from pods.
+func NewDetectLocalByBridgeInterfaceName(interfaceName string) (LocalTrafficDetector, error) {
+	if len(interfaceName) <= 0 {
+		return nil, fmt.Errorf("no bridge interface name set")
+	}
+	return &detectLocalByBridgeInterfaceName{
+		ifLocal:    []string{"-i", interfaceName},
+		ifNotLocal: []string{"!", "-i", interfaceName},
+	}, nil
+}
+
+func (d *detectLocalByBridgeInterfaceName) IsImplemented() bool {
+	return true
+}
+
+func (d *detectLocalByBridgeInterfaceName) IfLocal() []string {
+	return d.ifLocal
+}
+
+func (d *detectLocalByBridgeInterfaceName) IfNotLocal() []string {
+	return d.ifNotLocal
+}
+
+type detectLocalByInterfacePrefix struct {
+	ifLocal    []string
+	ifNotLocal []string
+}
+
+// NewDetectLocalByInterfacePrefix implements the LocalTrafficDetector interface using an interface prefix.
+// This can be used when a pod interface prefix can be used to capture the notion of local traffic. Note
+// that this will match on all interfaces that start with the given prefix.
+func NewDetectLocalByInterfacePrefix(interfacePrefix string) (LocalTrafficDetector, error) {
+	if len(interfacePrefix) <= 0 {
+		return nil, fmt.Errorf("no interface prefix set")
+	}
+	return &detectLocalByInterfacePrefix{
+		ifLocal:    []string{"-i", interfacePrefix + string('+')},
```

Comment:
This casting of rune to string is unusual and raises suspicion - why not `interfacePrefix + "+"` ?

Reply:
done.